File: app/src/main/java/com/gustavoforero/moviesomdb/net/OMDbRestClient.java
package com.gustavoforero.moviesomdb.net;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public enum OMDbRestClient {

    INSTANCE;

    private OMDbApi mOMDBOMDbApi;

    OMDbRestClient() {

        OkHttpClient client = new OkHttpClient.Builder()
                .connectTimeout(300, TimeUnit.SECONDS)
                .readTimeout(300, TimeUnit.SECONDS)
                .build();
        Gson gson = new GsonBuilder()
                .setLenient()
                .create();
        Retrofit retrofitService = new Retrofit.Builder().client(client)
                .baseUrl(OMDbApi.URL_BASE)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create(gson)).build();
        mOMDBOMDbApi = retrofitService.create(OMDbApi.class);
    }

    public OMDbApi getApi() {
        return mOMDBOMDbApi;
    }

}

File: app/src/main/java/com/gustavoforero/moviesomdb/net/OMDbRestClient.kt
package com.gustavoforero.moviesomdb.net

import com.google.gson.Gson
import com.google.gson.GsonBuilder

import java.util.concurrent.TimeUnit

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

enum class OMDbRestClient {
    INSTANCE;


    val omDbApi: OMDbApi

    init {
        val logging = HttpLoggingInterceptor()
        logging.level = HttpLoggingInterceptor.Level.BODY

        val client = OkHttpClient.Builder()
                .addInterceptor(logging)
                .connectTimeout(300, TimeUnit.SECONDS)
                .readTimeout(300, TimeUnit.SECONDS)
                .build()
        val gson = GsonBuilder()
                .setLenient()
                .create()
        val retrofitService = Retrofit.Builder().client(client)
                .baseUrl(OMDbApi.BASE_URL)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create(gson)).build()
        omDbApi = retrofitService.create(OMDbApi::class.java)
    }
}
